File: test/react-native/features/fixtures/scenario-launcher/src/scenarios/NativePromiseRejectionUnhandledScenario.js
import Scenario from './Scenario'
import { NativeInterface } from '../lib/native'

export class NativePromiseRejectionUnhandledScenario extends Scenario {
  async run () {
    await NativeInterface.runScenario('NativePromiseRejectionUnhandledScenario')
  }
}

File: test/react-native/features/fixtures/scenario-launcher/android/src/main/java/com/bugsnag/reactnative/test/scenarios/NativePromiseRejectionUnhandledScenario.kt
package com.reactnative.scenarios

import android.content.Context
import com.bugsnag.android.Bugsnag
import com.facebook.react.bridge.Promise

class NativePromiseRejectionUnhandledScenario(context: Context): Scenario(context) {

    override fun run(promise: Promise) {
        promise.reject(generateException())
    }
}
